backend/app/services/learning-service.py

```
@@ -0,0 +1,110 @@
+        self.config_service.learn_field_mapping(domain, field_info.get('name', ''), user_mapping, 1.0)
+
+    
+    def record_successfull_fill(self, domain: str, field_name: str, mapping: str, confidence: float):
+        self.config_service.learn_field_mapping(domain, field_name, mapping, confidence)
+
+    
+    def improve_field_classification(self, domain: str, field_info: Dict[str, Any]) -> Tuple[str, float]:
+        learned_mappings = self.config_service.get_learned_mappings(domain)
+
+        field_name = field_info.get('name', '').lower()
+        field_label = field_info.get('label', '').lower()
```

Ruff (0.12.2)

Local variable `field_label` is assigned to but never used

Remove assignment to unused variable `field_label`

(F841)